File: cdk-from-cfn-testing/expected/simple/golang/SimpleConstruct.go
package main

import (
	cdk "github.com/aws/aws-cdk-go/awscdk/v2"
	s3 "github.com/aws/aws-cdk-go/awscdk/v2/awss3"
	sqs "github.com/aws/aws-cdk-go/awscdk/v2/awssqs"
	"github.com/aws/constructs-go/constructs/v10"
	"github.com/aws/jsii-runtime-go"
)

type SimpleConstructProps struct {
	/// The prefix for the bucket name
	BucketNamePrefix *string
	LogDestinationBucketName interface{/* AWS::SSM::Parameter::Value<String> */}
	/// A number parameter to test type generation
	NumberParam *float64
}

/// An example stack that uses many of the syntax elements permitted in a
/// CloudFormation template, but does not attempt to represent a realistic stack.
type SimpleConstruct struct {
	constructs.Construct
	/// The ARN of the bucket in this template!
	BucketArn interface{} // TODO: fix to appropriate type
	/// The ARN of the SQS Queue
	QueueArn interface{} // TODO: fix to appropriate type
	/// Whether this is a large region or not
	IsLarge interface{} // TODO: fix to appropriate type
}

func NewSimpleConstruct(scope constructs.Construct, id string, props *SimpleConstructProps) *SimpleConstruct {
	/*
	booleans := map[*string]map[*string]*bool{
		jsii.String("True"): map[*string]*bool{
			jsii.String("true"): jsii.Bool(true),
		},
		jsii.String("False"): map[*string]*bool{
			jsii.String("false"): jsii.Bool(false),
		},
	}
	*/

	/*
	lists := map[*string]map[*string][]*string{
		jsii.String("Candidates"): map[*string][]*string{
			jsii.String("Empty"): []*string{
			},
			jsii.String("Singleton"): []*string{
				jsii.String("One"),
			},
			jsii.String("Pair"): []*string{
				jsii.String("One"),
				jsii.String("Two"),
			},
		},
	}
	*/

	/*
	numbers := map[*string]map[*string]*float64{
		jsii.String("Prime"): map[*string]*float64{
			jsii.String("Eleven"): jsii.Number(11),
			jsii.String("Thirteen"): jsii.Number(13),
			jsii.String("Seventeen"): jsii.Number(17),
		},
	}
	*/

	strings := map[*string]map[*string]*string{
		jsii.String("Foos"): map[*string]*string{
			jsii.String("Foo1"): jsii.String("Foo1"),
			jsii.String("Foo2"): jsii.String("Foo2"),
		},
		jsii.String("Bars"): map[*string]*string{
			jsii.String("Bar"): jsii.String("Bar"),
		},
	}

	table := map[*string]map[*string]interface{}{
		jsii.String("Values"): map[*string]interface{}{
			jsii.String("Boolean"): jsii.Bool(true),
			jsii.String("Float"): jsii.Number(3.14),
			jsii.String("List"): []*string{
				jsii.String("1"),
				jsii.String("2"),
				jsii.String("3"),
			},
			jsii.String("Number"): jsii.Number(42),
			jsii.String("String"): jsii.String("Baz"),
		},
	}

	construct := constructs.NewConstruct(scope, &id)

	isUs := cdk.Fn_Select(jsii.Number(0), cdk.Fn_Split(jsii.String("-"), cdk.Stack_Of(construct).Region())) == jsii.String("us")

	isUsEast1 := cdk.Stack_Of(construct).Region() == jsii.String("us-east-1")

	isLargeRegion := isUsEast1

	queue := sqs.NewCfnQueue(
		construct,
		jsii.String("Queue"),
		&sqs.CfnQueueProps{
			DelaySeconds: jsii.Number(42),
			SqsManagedSseEnabled: jsii.Bool(false),
			KmsMasterKeyId: cdk.Fn_ImportValue(jsii.String("Shared-KmsKeyArn")),
			QueueName: cdk.Fn_Join(jsii.String("-"), &[]*string{
				cdk.Stack_Of(construct).StackName(),
				strings[jsii.String("Bars")][jsii.String("Bar")],
				cdk.Fn_Select(jsii.Number(1), cdk.Fn_GetAzs(cdk.Stack_Of(construct).Region())),
			}),
			RedrivePolicy: nil,
			VisibilityTimeout: jsii.Number(120),
		},
	)

	bucket := s3.NewCfnBucket(
		construct,
		jsii.String("Bucket"),
		&s3.CfnBucketProps{
			AccessControl: jsii.String("Private"),
			LoggingConfiguration: &LoggingConfigurationProperty{
				DestinationBucketName: props.LogDestinationBucketName,
			},
			WebsiteConfiguration: &WebsiteConfigurationProperty{
				RedirectAllRequestsTo: &RedirectAllRequestsToProperty{
					HostName: jsii.String("example.com"),
					Protocol: jsii.String("https"),
				},
			},
			Tags: &[]*cdk.CfnTag{
				&cdk.CfnTag{
					Key: jsii.String("FancyTag"),
					Value: ifCondition(
						isUsEast1,
						cdk.Fn_Base64(table[jsii.String("Values")][jsii.String("String")]),
						cdk.Fn_Base64(jsii.String("8CiMvAo=")),
					),
				},
			},
		},
	)

	cdk.NewCfnOutput(construct, jsii.String("CfnOutputBucketArn"), &cdk.CfnOutputProps{
		Key: jsii.String("BucketArn"),
		Description: jsii.String("The ARN of the bucket in this template!"),
		ExportName: jsii.String("ExportName"),
		Value: bucket.AttrArn(),
	})

	return &SimpleConstruct{
		Construct: construct,
		BucketArn: bucket.AttrArn(),
		QueueArn: queue.Ref(),
		IsLarge: ifCondition(
			isLargeRegion,
			jsii.Bool(true),
			jsii.Bool(false),
		),
	}
}

/// ifCondition is a helper function that replicates the ternary
/// operator that can be found in other languages. It is conceptually
/// equivalent to writing `cond ? whenTrue : whenFalse`, meaning it
/// returns `whenTrue` if `cond` is `true`, and `whenFalse` otherwise.
func ifCondition[T any](cond bool, whenTrue T, whenFalse T) T {
	if cond {
		return whenTrue
	}
	return whenFalse
}

File: cdk-from-cfn-testing/expected/simple/python/SimpleConstruct.py
from aws_cdk import Stack
import aws_cdk as cdk
import aws_cdk.aws_s3 as s3
import aws_cdk.aws_sqs as sqs
from constructs import Construct
import base64

"""
  An example stack that uses many of the syntax elements permitted in a
  CloudFormation template, but does not attempt to represent a realistic stack.
"""
class SimpleConstruct(Construct):
  def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
    super().__init__(scope, construct_id)

    # Applying default props
    props = {
      'bucketNamePrefix': kwargs.get('bucketNamePrefix', 'bucket'),
      'logDestinationBucketName': cdk.CfnParameter(self, 'logDestinationBucketName', 
        type = 'AWS::SSM::Parameter::Value<String>',
        default = str(kwargs.get('logDestinationBucketName', '/logging/bucket/name')),
      ),
      'numberParam': kwargs.get('numberParam', 42),
    }

    # Mappings
    booleans = {
      'True': {
        'true': True,
      },
      'False': {
        'false': False,
      },
    }
    lists = {
      'Candidates': {
        'Empty': [],
        'Singleton': ['One'],
        'Pair': ['One','Two'],
      },
    }
    numbers = {
      'Prime': {
        'Eleven': 11,
        'Thirteen': 13,
        'Seventeen': 17,
      },
    }
    strings = {
      'Foos': {
        'Foo1': 'Foo1',
        'Foo2': 'Foo2',
      },
      'Bars': {
        'Bar': 'Bar',
      },
    }
    table = {
      'Values': {
        'Boolean': True,
        'Float': 3.14,
        'List': ['1','2','3'],
        'Number': 42,
        'String': 'Baz',
      },
    }

    # Conditions
    is_us = cdk.Fn.select(0, Stack.of(self).region.split('-')) == 'us'
    is_us_east1 = Stack.of(self).region == 'us-east-1'
    is_large_region = is_us_east1

    # Resources
    queue = sqs.CfnQueue(self, 'Queue',
          delay_seconds = 42,
          sqs_managed_sse_enabled = False,
          kms_master_key_id = cdk.Fn.import_value('Shared-KmsKeyArn'),
          queue_name = '-'.join([
            Stack.of(self).stack_name,
            strings['Bars']['Bar'],
            cdk.Fn.select(1, cdk.Fn.get_azs(Stack.of(self).region)),
          ]),
          redrive_policy = None,
          visibility_timeout = 120,
        )
    queue.cfn_options.deletion_policy = cdk.CfnDeletionPolicy.RETAIN_EXCEPT_ON_CREATE

    bucket = s3.CfnBucket(self, 'Bucket',
          access_control = 'Private',
          logging_configuration = {
            'destinationBucketName': props['logDestinationBucketName'],
          },
          website_configuration = {
            'redirectAllRequestsTo': {
              'hostName': 'example.com',
              'protocol': 'https',
            },
          },
          tags = [
            {
              'key': 'FancyTag',
              'value': cdk.Fn.base64(table['Values']['String']) if is_us_east1 else base64.b64decode('8CiMvAo='),
            },
          ],
        ) if is_us_east1 else None
    if (bucket is not None):
      bucket.cfn_options.metadata = {
        'CostCenter': 1337,
      }
      bucket.cfn_options.deletion_policy = cdk.CfnDeletionPolicy.DELETE
      bucket.add_dependency(queue)

    # Outputs
    """
      The ARN of the bucket in this template!
    """
    self.bucket_arn = bucket.attr_arn if is_us_east1 else None
    if (is_us_east1):
      cdk.CfnOutput(self, 'CfnOutputBucketArn', 
        key = 'BucketArn',
        description = 'The ARN of the bucket in this template!',
        export_name = 'ExportName',
        value = str(self.bucket_arn),
      )


    """
      The ARN of the SQS Queue
    """
    self.queue_arn = queue.ref
    cdk.CfnOutput(self, 'CfnOutputQueueArn', 
      key = 'QueueArn',
      description = 'The ARN of the SQS Queue',
      value = str(self.queue_arn),
    )

    """
      Whether this is a large region or not
    """
    self.is_large = True if is_large_region else False
    cdk.CfnOutput(self, 'CfnOutputIsLarge', 
      key = 'IsLarge',
      description = 'Whether this is a large region or not',
      value = str(self.is_large),
    )



